lib/src/combined_wrappers/combined_list.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+/// Returns a new list that represents [lists] flattened into a single list.
+///
+/// All methods and accessors treat the new list as-if it were a single
+/// concatenated list, but the underlying implementation is based on lazily
+/// accessing individual list instances.
+///
+/// The resulting list has an index operator (`[]`) and `length` property that
+/// are both `O(lists)`, rather than `O(1)`, and the list is unmodifiable.
+List/*<T>*/ combineLists/*<T>*/(List<List/*<T>*/> lists) {
+  // Small optimization when there are no lists to avoid allocation.
+  if (lists.isEmpty) {
+    return const [];
```

Comment:
Hmm. I agree with this. Changed it up so the underlying list can always be changed.